Pop/PopObjectConverter.cs

```
@@ -59,11 +60,19 @@ public static object DeserializeFromPopObject(string jsonMessage)
                     case JTokenType.Property:
                     case JTokenType.Comment:
                     case JTokenType.Raw:
-                        throw new NotImplementedException();
+                        throw new NotSupportedException($"Type of:{type.Name}  Is Not Supported");
                     default:
                         throw new ArgumentOutOfRangeException();
                 }
             }
+            catch (ArgumentOutOfRangeException)
```

We probably don't need to re-throw this exception.